grails-app/services/org/broadinstitute/orsp/ConsentService.groovy

```diff
@@ -513,6 +514,26 @@ class ConsentService implements Status {
         }
     }
 
+    /**
+     * Find all consent groups associated to the specified consent collection links
+     *
+     * @param keys The consent group key and related consent collection link
+     * @return List of Consent Group that are related to the specified consent collection links
+     */
+    Collection<Issue> findConsentGroupsAssociatedByConsentLink(Map<String, ConsentCollectionLink> keys) {
+        if (keys && !keys.isEmpty()) {
+            List<Issue> issues = Issue.findAllByProjectKeyInList(keys.keySet().toList())
+            issues.each { issue ->
+                // set all documents related to the specified consent group including its consent collection link documents
+                issue.setAttachments(queryService.getAttachmentsForProjects(keys.keySet()))
```

Comment:
I strive to avoid making dao calls inside loops. This leads to all sorts of slowness. It's better to make a new `QueryService` method that queries for all docs on a list of project keys and retrieve them as a map of project key -> list of docs. Then loop over the map and assign the docs to the issue.

Reply:
I agree with you, I didn't want to refactor the existing code, I just only moved the method and I tried only to change documents logic. But it's a good opportunity to improve the existing code